### PR TITLE
Reduce number of dependabot PRs

### PR DESCRIPTION
**Pull request by glatterf42**
This PR adapts the dependabot config file to reduce the number of PRs created by bumping versions. It does so in two ways:

1. [Group all dependencies together](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/customizing-dependency-updates#example-2) so that version updates will only be one PR
2. Decrease the [`schedule.interval`](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#scheduleinterval) to monthly

## How to review

- Read the diff and note that the CI checks all pass.

## PR checklist

- [x] Continuous integration checks all ✅
- ~[ ] Add or expand tests; coverage checks both ✅~ No functionality added.
- ~[ ] Add, expand, or update documentation.~ Nothing user-facing added.
- ~[ ] Update release notes.~ Nothing user-facing added.

### CODE REVIEW

**Comment by glatterf42** [comment]
@khaeru, If you don't find time until Friday to review this, I will just merge this without approval since next dependabot PR would be due next Monday with our current configuration.

**Review by khaeru — APPROVED** [review]
Looks good; LGTM.

**Pull request merged**
